src/test/java/com/scottlogic/librarygradproject/cucumber/features/edit_book.feature

```diff
@@ -0,0 +1,40 @@
+Feature: Correctly editing books in the database
+
+  Background:
+    Given a filled book repository exists
+
+  Scenario: An edit book request is received with all fields correctly filled
+    When An edit book request is received with correct book details
+    Then the modified book should replace the original
+
+  Scenario: An edit book request is received with incorrect ISBN value
+    When An edit book request is received with incorrect ISBN
+    Then the database should not be modified
+
+  Scenario: An edit book request is received without author value
+    When An edit book request is received without author
+    Then the database should not be modified
+
+  Scenario: An edit book request is received without title value
+    When An edit book request is received without title
+    Then the database should not be modified
+
+  Scenario: An edit book request is received with author too long
+    When An edit book request is received with author too long
+    Then the database should not be modified
+
+  Scenario: An edit book request is received with title too long
+    When An edit book request is received with title too long
+    Then the database should not be modified
+
+  Scenario: An edit book request is received with incorrect date format
+    When An edit book request is received with incorrect date format
+    Then the database should not be modified
+
+  Scenario: An edit book request is received with leading and trailing whitespace in fields but otherwise acceptable
+    When An edit book request is received with leading and trailing whitespace in fields otherwise acceptable
+    And the modified book should replace the original
+
+  Scenario: An edit book request is received with leading and trailing whitespace in fields and otherwise unacceptable
+    When An edit book request is received with leading and trailing whitespace in fields and unacceptable
+    And the database should not be modified
```

Comment:
Maybe you could add test cases for: 
- Editing a book to have no ISBN 
- Editing a book to have no publish date 
And vice versa so adding ISBN/ Publish date to books that currently don't have them (this would more likely be the common practice)

Reply:
Added these tests now to both edit & add